### ISSUE TITLE
Consider making `take`, `drop`, ... curried on streams

### ISSUE THREAD

**Issue by lostintime** [mikearnaldi/matechs-effect#137]
Hi and thanks for sharing this library, it looks awesome ...

I was giving it a try and found that `stream.take` function  (and others) take 2 arguments, so id doesn't look too nice with `pipe`:

```typescript
import { effect as T, stream as S } from "@matechs/effect"
import { pipe } from "fp-ts/lib/pipeable"

pipe(
  S.periodically(100),
  S.chain(n => S.encaseEffect(T.pure(n))),
  s => S.take(s, 100) // -> S.take(100)
)
```

Did you consider changing signature to `(n: number) => <S>(stream: S) => ...`? If so - what are the drawbacks of doing so?

Thanks!

**Comment by mikearnaldi** [comment]
Thanks for spotting it, they should definately be curried

**Comment by mikearnaldi** [comment]
released in v3.0.0 (major simply because non backward compatible at the api level)

**Comment by lostintime** [comment]
Yeah, definitely a breaking change ..

Do you keep tracking of such issues in other place? It would be nice to be able to monitor the progress in a public place (a github milestone or something ..)
